internal/controllers/reconciliation/controller.go

```diff
@@ -146,17 +138,49 @@ func (c *Controller) Reconcile(ctx context.Context, req *reconstitution.Request)
 		return ctrl.Result{}, fmt.Errorf("getting resource slice: %w", err)
 	}
 	var ready *metav1.Time
-	if status := resource.FindStatus(slice); status == nil || status.Ready == nil {
+	status := resource.FindStatus(slice)
+	if status == nil || status.Ready == nil {
 		readiness, ok := resource.ReadinessChecks.EvalOptionally(ctx, current)
 		if ok {
 			ready = &readiness.ReadyTime
 		}
+	} else {
+		ready = status.Ready
+	}
+
+	// Evaluate the readiness of resources in the previous readiness group
+	if (status == nil || !status.Reconciled) && !resource.Deleted() {
+		dependencies := c.resourceClient.RangeByReadinessGroup(ctx, synRef, resource.ReadinessGroup, -1)
```

Comment:
Correct me if I'm wrong, but the way I'm reading this and the indexing PR, these are just dependencies within the same composition; will we also have readiness groups across compositions?

Reply:
I totally see the use of intra-composition dependencies (get etcd ready before api-server in control-plane-core for ex) but we should also run through the possible scenarios where a composition might depend on another one

Reply:
This is an interesting thought - let's discuss elsewhere

Reply:
Discussed offline, will stick with intra-composition readiness for now, we'll evaluate if inter-composition is required in the future